tests/storage/export.go

```diff
@@ -913,6 +913,48 @@ var _ = SIGDescribe("Export", func() {
 		Expect(*export.Status.TokenSecretRef).ToNot(BeEmpty())
 	})
 
+	It("Should honor TTL by cleaning up the the VMExport altogether", func() {
+		sc, exists := libstorage.GetRWOFileSystemStorageClass()
+		if !exists {
+			Skip("Skip test when Filesystem storage is not present")
+		}
+
+		pvc, _ := populateKubeVirtContent(sc, k8sv1.PersistentVolumeFilesystem)
+		ttl := 2 * time.Minute
+		export := &exportv1.VirtualMachineExport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("test-export-%s", rand.String(12)),
+				Namespace: pvc.Namespace,
+			},
+			Spec: exportv1.VirtualMachineExportSpec{
+				Source: k8sv1.TypedLocalObjectReference{
+					APIGroup: &k8sv1.SchemeGroupVersion.Group,
+					Kind:     "PersistentVolumeClaim",
+					Name:     pvc.Name,
+				},
+				TTLSecondsAfterCreation: pointer.Int32Ptr(int32(ttl.Seconds())),
+			},
+		}
+		export, err := virtClient.VirtualMachineExport(export.Namespace).Create(context.Background(), export, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		By("Making sure the export becomes ready")
+		waitForReadyExport(export)
+
+		// VMExport sticks around exactly until TTL expiration time is reached
+		Consistently(func() error {
+			_, err := virtClient.VirtualMachineExport(export.Namespace).Get(context.Background(), export.Name, metav1.GetOptions{})
+			return err
+		}, ttl, 10*time.Second).Should(Succeed())
```

Comment:
maybe Im missing something but the exparation is counted since the export is created+ttl, and you start this check after export is ready- not created, so some timed passed between creating the export and it being ready, so isnt it possible that this check is a bit racy if the export takes more then 10 seconds to be ready (then the last check in this Consistently might fail?)

Reply:
Totally right

Reply:
Done